src/pyOpenMS/pyopenms/_dataframes.py

```diff
@@ -330,7 +330,7 @@ def get_df(self, long : bool = False):
         if long:
             cols = ["RT", "mz", "inty"]
             self.updateRanges()
-            spectraarrs2d = self.get2DPeakDataLong(self.getMinRT(), self.getMaxRT(), self.getMinMZ(), self.getMaxMZ())
+            spectraarrs2d = self.get2DPeakDataLong(self.getMinRT(), self.getMaxRT(), self.getMinMZ(), self.getMaxMZ(), 1)
```

Comment:
@axelwalter the get2DPeakDataLong has now the parameter to extract a certain MS level. Did we in the past extract all peaks? And if so, how did we distinguish between e.g. MS1 and MS2?

Reply:
also @t0mdavid-m and @jcharkow is extracting all peaks irrespective of MS level a use case? Do we need this somewhere?

Reply:
I, personally, set this during mzml load.
I hope others do this as well

Reply:
> also @t0mdavid-m and @jcharkow is extracting all peaks irrespective of MS level a use case? Do we need this somewhere?

I was hoping for the getDataAll or the Spectrum based functions to return a DF with the level annotated.
Alternatively we can provide a range of levels to extract also.

Reply:
Is this really needed? e.g., we would ideally also need at least the native identifier or even the type (spectrum or chromatogram) if want to extract all data and really make sense of such a table. Is that maybe a bit too much?

Reply:
E.g. for plotting a 2D peak map the option to get the DataFrame in long format is very useful. The addition of MS level selection is great! In the past I simply added a MS level column manually. I never had a use case where a native ID was required.

Reply:
No strong opinion. This would only be for spectra. Yes, native ID could be another column.